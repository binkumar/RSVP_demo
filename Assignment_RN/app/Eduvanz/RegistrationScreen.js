import React, { Component } from 'react';
import { View, Text, StyleSheet, TextInput, KeyboardAvoidingView, ActivityIndicator, TouchableHighlight, ScrollView, Alert } from 'react-native';
import DatePicker from 'react-native-datepicker'
import SegmentedControlTab from "react-native-segmented-control-tab";
import { connect } from 'react-redux';

import { apiKeys } from '../Redux/Actions/Helper/apiKeys';
import * as actions from '../Redux/Actions';
import HamburgerMenu from './Component/HamburgerMenu';
/*
First Screen
It will have a registration form with following fields â€”
Name
Age
D.O.B (JS Date object)
Profession (can be Employed/Student)
Locality
Number of Guests (0-2)
Address (multiline input upto 50 characters)
With a submit button you can simply mock a submit API call with services like Beeceptor. 
Also ensure you show an alert to the user if they click back while editing the form.
*/
const LABEL_TYPE = {
    NAME: 'Name',
    AGE: "Age",
    DOB: "Dob",
    PROFESSION: "Profession",
    LOCALITY: "Locality",
    GUESTS: "Guests",
    ADDRESS: "Address"
};

class RegistrationScreen extends Component {

    constructor(props) {
        super(props);
        this.state = {
            name: "",
            age: 0,
            dob: null,
            profession: 'Employed',
            locality: null,
            no_of_guests: 0,
            address: null,
            selectedIndexProf: 0,
            selectedIndexGuest: 0
        };
    }

    componentDidUpdate(prevProps, prevState) {
        if (prevProps.registration !== this.props.registration && !this.props.registration.isLoading) {
            if (this.props.error) {
                this.createTwoButtonAlert("Registration Failed");
            } else {
                this.createTwoButtonAlert("Registration Success");
            }
        }
    }

    validation = () => {
        for (const [key, value] of Object.entries(LABEL_TYPE)) {
            if (!this.state[value]) {
                return false;
            }
        }
        return true;
    };

    createTwoButtonAlert = (message) =>
        Alert.alert(
            "",
            message,
            [
                {
                    text: "Cancel",
                    onPress: () => console.log("Cancel Pressed"),
                    style: "cancel"
                },
                { text: "OK", onPress: () => console.log("OK Pressed") }
            ],
            { cancelable: false }
        );

    createInputView = (labelType, isMultline = false, keyboardType = 'default') => (
        <View style={{ height: 50, marginHorizontal: 10, marginVertical: 10 }}>
            <Text style={{ height: 20, fontSize: 15 }}>{labelType}:</Text>
            <TextInput
                style={{
                    height: labelType === LABEL_TYPE.ADDRESS ? 100 : 35,
                    backgroundColor: 'white',
                    marginTop: 4,
                    borderRadius: 5,
                    padding: 10,
                    alignItems: 'flex-start',
                    justifyContent: 'center',
                    shadowColor: "#000",
                    shadowOffset: {
                        width: 0,
                        height: 2
                    },
                    shadowOpacity: 0.25,
                    shadowRadius: 3.84,
                    elevation: 5
                }}
                autoFocus={true}
                keyboardType={keyboardType}
                multiline={isMultline}
                clearTextOnFocus={false}
                placeholderTextColor={'grey'}
                placeholder={`Enter ${labelType}`}
                underlineColorAndroid="transparent"
                value={this.state.searchText}
                onChangeText={text => {
                    this.setState({ [labelType]: text });
                }}
            />
        </View>
    )

    createSegementControl = (labelType) => {
        let segementData = null;
        let selectedIndex = 0;
        if (labelType === LABEL_TYPE.PROFESSION) {
            segementData = ['Employed', 'Student'];
            selectedIndex = this.state.selectedIndexProf;
        } else if (labelType === LABEL_TYPE.GUESTS) {
            segementData = ["0", "1", "2"];
            selectedIndex = this.state.selectedIndexGuest;
        }
        return (
            <View style={{ height: 50, marginHorizontal: 10, marginVertical: 10 }}>
                <Text style={{ height: 20, fontSize: 15 }}>{labelType}:</Text>
                <SegmentedControlTab
                    values={segementData}
                    selectedIndex={selectedIndex}
                    onTabPress={index => {
                        this.setState({ [labelType]: segementData[index] });
                        if (labelType === LABEL_TYPE.PROFESSION) {
                            this.setState({ selectedIndexProf: index });
                        } else if (labelType === LABEL_TYPE.GUESTS) {
                            this.setState({ selectedIndexGuest: index });
                        }
                    }}
                />
            </View>
        )
    };

    createDatePicker = (labelType) => (
        <View style={{ height: 50, marginHorizontal: 10, marginVertical: 10 }}>
            <Text style={{ height: 20, fontSize: 15 }}>{labelType}:</Text>
            <DatePicker
                style={{ width: 200, backgroundColor: 'white' }}
                date={this.state[labelType]}
                mode="date"
                placeholder="select date"
                format="YYYY-MM-DD"
                // minDate="2016-05-01"
                // maxDate="2016-06-01"
                confirmBtnText="Confirm"
                cancelBtnText="Cancel"
                customStyles={{
                    dateIcon: {
                        backgroundColor: 'white',
                        position: 'absolute',
                        left: 0,
                        top: 4,
                        marginLeft: 0
                    },
                    dateInput: {
                        marginLeft: 36
                    }
                }}
                onDateChange={(date) => { this.setState({ [labelType]: date }) }}
            />
        </View>
    )

    createSubmitBtn = () => (<TouchableHighlight
        onPress={() => {
            if (this.validation()) {
                this.props.fetchContent(apiKeys.registration)
            } else {
                this.createTwoButtonAlert("Incomplete informations");
            }
        }
        }
        style={styles.btn}>
        <Text
            style={{ fontSize: 14, marginTop: 5, color: 'white' }}>
            Submit
            </Text>
    </TouchableHighlight>);

    render() {
        return (
            <KeyboardAvoidingView
                behavior={'height'}
                style={styles.container}
            // keyboardVerticalOffset={Platform.OS === "ios" ? 60 : 0}
            >
                <HamburgerMenu
                    screenTitle={'Registration'}
                    onMenuPress={() => this.props.navigation.openDrawer()} />
                <ScrollView>
                    <View style={{ flex: 1, justifyContent: 'flex-start' }}>
                        {this.createInputView(LABEL_TYPE.NAME)}
                        {this.createInputView(LABEL_TYPE.AGE, false, 'numeric')}
                        {this.createDatePicker(LABEL_TYPE.DOB)}
                        {this.createSegementControl(LABEL_TYPE.PROFESSION)}
                        {this.createInputView(LABEL_TYPE.LOCALITY)}
                        {this.createSegementControl(LABEL_TYPE.GUESTS)}
                        {this.createInputView(LABEL_TYPE.ADDRESS, true)}
                        {this.props.registration.isLoading && <ActivityIndicator color='black' size='large' />}
                        {this.createSubmitBtn()}
                    </View>
                </ScrollView>
            </KeyboardAvoidingView>
        );
    }
}
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'white'
    },
    btn: {
        height: 40,
        marginHorizontal: 10,
        backgroundColor: 'blue',
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 100,
        borderRadius: 20,
        padding: 10,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5
    }
});

function mapStateToProps(state) {
    return {
        registration: state.registration,
    };
}

export default connect(
    mapStateToProps,
    actions,
)(RegistrationScreen);