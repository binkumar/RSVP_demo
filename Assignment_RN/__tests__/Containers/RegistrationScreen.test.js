import 'react-native';
import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import RegistrationScreen from '../../app/Eduvanz/RegistrationScreen';

const mockStore = configureStore();

const navigation = {
  dispatch: jest.fn(),
};

const mockState = { registration: {} };

describe('RegistrationScreen', () => {
  it('renders correctly ', () => {
    const store = mockStore(mockState);
    const tree = renderer.create(
      <Provider store={store}>
        <RegistrationScreen navigation={navigation} />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
