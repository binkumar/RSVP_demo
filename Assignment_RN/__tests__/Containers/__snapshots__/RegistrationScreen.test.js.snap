// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationScreen renders correctly  1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
    testID="headerContainer"
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï€¹
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Registration
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "height": 50,
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "height": 20,
              }
            }
          >
            Name
            :
          </Text>
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            clearTextOnFocus={false}
            keyboardType="default"
            multiline={false}
            onChangeText={[Function]}
            placeholder="Enter Name"
            placeholderTextColor="grey"
            rejectResponderTermination={true}
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderRadius": 5,
                "elevation": 5,
                "height": 35,
                "justifyContent": "center",
                "marginTop": 4,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "height": 50,
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "height": 20,
              }
            }
          >
            Age
            :
          </Text>
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            clearTextOnFocus={false}
            keyboardType="numeric"
            multiline={false}
            onChangeText={[Function]}
            placeholder="Enter Age"
            placeholderTextColor="grey"
            rejectResponderTermination={true}
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderRadius": 5,
                "elevation": 5,
                "height": 35,
                "justifyContent": "center",
                "marginTop": 4,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "height": 50,
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "height": 20,
              }
            }
          >
            Dob
            :
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "width": 142,
                },
                Object {
                  "backgroundColor": "white",
                  "width": 200,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#aaa",
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    Object {
                      "marginLeft": 36,
                    },
                    false,
                    false,
                  ]
                }
              >
                <Text
                  allowFontScaling={true}
                  style={
                    Array [
                      Object {
                        "color": "#c9c9c9",
                      },
                      undefined,
                    ]
                  }
                >
                  select date
                </Text>
              </View>
              <Image
                source={
                  Object {
                    "testUri": "../../react-native-datepicker/date_icon.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 32,
                      "marginLeft": 5,
                      "marginRight": 5,
                      "width": 32,
                    },
                    Object {
                      "backgroundColor": "white",
                      "left": 0,
                      "marginLeft": 0,
                      "position": "absolute",
                      "top": 4,
                    },
                  ]
                }
              />
              <Modal
                animationType="none"
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                supportedOrientations={
                  Array [
                    "portrait",
                    "portrait-upside-down",
                    "landscape",
                    "landscape-left",
                    "landscape-right",
                  ]
                }
                transparent={true}
                visible={false}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "backgroundColor": "#00000077",
                        "flex": 1,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={false}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "height": 0,
                            "overflow": "hidden",
                          }
                        }
                      >
                        <View
                          pointerEvents="auto"
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "borderTopColor": "#ccc",
                                  "borderTopWidth": 1,
                                  "marginTop": 42,
                                },
                                undefined,
                              ]
                            }
                          >
                            <RCTDatePicker
                              date={1603277875274}
                              mode="date"
                              onChange={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "height": 216,
                                }
                              }
                              timeZoneOffsetInMinutes={null}
                            />
                          </View>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 42,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                                "position": "absolute",
                                "top": 0,
                              },
                              Object {
                                "left": 0,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={true}
                            style={
                              Array [
                                Object {
                                  "color": "#46cf98",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#666",
                                },
                                undefined,
                              ]
                            }
                          >
                            Cancel
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 42,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                                "position": "absolute",
                                "top": 0,
                              },
                              Object {
                                "right": 0,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={true}
                            style={
                              Array [
                                Object {
                                  "color": "#46cf98",
                                  "fontSize": 16,
                                },
                                undefined,
                              ]
                            }
                          >
                            Confirm
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </Modal>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 50,
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "height": 20,
              }
            }
          >
            Profession
            :
          </Text>
          <View
            removeClippedSubviews={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "flexDirection": "row",
                },
                Object {},
              ]
            }
          >
            <View
              accessibilityLabel="Employed"
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0076FF",
                  "borderBottomLeftRadius": 5,
                  "borderColor": "#0076FF",
                  "borderRightWidth": 1,
                  "borderTopLeftRadius": 5,
                  "borderWidth": 1,
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#0076FF",
                      },
                      Object {},
                      Array [
                        Object {
                          "color": "white",
                        },
                        Object {},
                      ],
                    ]
                  }
                >
                  Employed
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="Student"
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomRightRadius": 5,
                  "borderColor": "#0076FF",
                  "borderLeftWidth": 0,
                  "borderTopRightRadius": 5,
                  "borderWidth": 1,
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#0076FF",
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  Student
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 50,
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "height": 20,
              }
            }
          >
            Locality
            :
          </Text>
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            clearTextOnFocus={false}
            keyboardType="default"
            multiline={false}
            onChangeText={[Function]}
            placeholder="Enter Locality"
            placeholderTextColor="grey"
            rejectResponderTermination={true}
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderRadius": 5,
                "elevation": 5,
                "height": 35,
                "justifyContent": "center",
                "marginTop": 4,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "height": 50,
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "height": 20,
              }
            }
          >
            Guests
            :
          </Text>
          <View
            removeClippedSubviews={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "flexDirection": "row",
                },
                Object {},
              ]
            }
          >
            <View
              accessibilityLabel="0"
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0076FF",
                  "borderBottomLeftRadius": 5,
                  "borderColor": "#0076FF",
                  "borderRightWidth": 0,
                  "borderTopLeftRadius": 5,
                  "borderWidth": 1,
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#0076FF",
                      },
                      Object {},
                      Array [
                        Object {
                          "color": "white",
                        },
                        Object {},
                      ],
                    ]
                  }
                >
                  0
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="1"
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderColor": "#0076FF",
                  "borderWidth": 1,
                  "flex": 1,
                  "justifyContent": "center",
                  "marginLeft": -1,
                  "opacity": 1,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#0076FF",
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  1
                </Text>
              </View>
            </View>
            <View
              accessibilityLabel="2"
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomRightRadius": 5,
                  "borderColor": "#0076FF",
                  "borderLeftWidth": 0,
                  "borderTopRightRadius": 5,
                  "borderWidth": 1,
                  "flex": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#0076FF",
                      },
                      Object {},
                      Object {},
                    ]
                  }
                >
                  2
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 50,
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "height": 20,
              }
            }
          >
            Address
            :
          </Text>
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            clearTextOnFocus={false}
            keyboardType="default"
            multiline={true}
            onChangeText={[Function]}
            placeholder="Enter Address"
            placeholderTextColor="grey"
            rejectResponderTermination={true}
            style={
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "white",
                "borderRadius": 5,
                "elevation": 5,
                "height": 100,
                "justifyContent": "center",
                "marginTop": 4,
                "padding": 10,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "blue",
              "borderRadius": 20,
              "elevation": 5,
              "height": 40,
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginTop": 100,
              "padding": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
                "marginTop": 5,
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
